pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HttpSegmentFetcher.java

```diff
@@ -50,6 +52,9 @@ public void fetchSegmentToLocal(String uri, File tempFile) throws Exception {
             "Downloaded file from {} to {}; Length of httpGetResponseContent: {}; Length of downloaded file: {}", uri,
             tempFile, httpGetResponseContentLength, tempFile.length());
         return;
+      } catch (PermanentDownloadException e) {
+        LOGGER.error("Failed to download file from {}", uri, e);
+        Utils.rethrowException(e);
```

Comment:
Need a metric to alert on missing segments ?

Reply:
we have the segment status checker bumping metrics for error segments, so we should be ok